wallet/src/de/schildbach/wallet/ui/backup/BackupWalletDialogFragment.java

```diff
@@ -92,11 +96,14 @@ public static void show(final FragmentManager fm) {
     private View passwordMismatchView;
     private CheckBox showView;
     private TextView warningView;
+    private View spendingPINViewGroup;
+    private EditText spendingPINView;
+    private TextView badSpendingPINView;
     private Button positiveButton, negativeButton;
-
+    private Executor executor = Executors.newSingleThreadExecutor();
+    private Wallet backupWallet;
```

Comment:
I think the (decrypted) backupWallet should be kept in a `MutableLiveData<Wallet>` in `BackupWalletViewModel`, rather than here where is can be collected at any time (e.g. while you're picking the file).

And I'd use the name `walletToBackUp`, maybe even `decryptedWalletToBackUp`.